### DISCUSSION TITLE
Has anyone got a 1.3" SH1106 Oled working on LBB?

### DISCUSSION THREAD

**Discussion by estaban-zarzamora** [outdoorbits/little-backup-box#196] [Q&A]
Has anyone got a 1.3" SH1106 Oled working on LBB? A SSD1306 works fine, SH1106 shows static / snow on the display.

**Comment by estaban-zarzamora** [comment]
...or otherwise a 128x32 instead of a 128x64 SSD1306 OLED?

**Comment by outdoorbits** [comment]
I didn't test it, but if the 32 lines display works at all, most probably it will not display the progress of your backup.

**Comment by estaban-zarzamora** [comment]
Got the 128x32 i2c SSD1306 OLED working:
https://user-images.githubusercontent.com/47415283/219853966-7ea79677-4d52-4b3a-827b-5195cc3de624.MOV

I changed the "disp = adafruit_ssd1306.SSD1306_I2C(128, 64, i2c,addr=I2C_ADDRESS)" to 32 in oled.py, but doesn't change anything. Also lowering the font size in oled.py (line 84) doesn't seem to change anything.

Is there a way to have better readability? (and don't tell me to go for a 128x64 ;-) please...)

**Comment by outdoorbits** [comment]
I can hear a voice but there is no image. Could you post a image please?

**Comment by estaban-zarzamora** [comment]
![2023-02-18 14_21_44-](https://user-images.githubusercontent.com/47415283/219868228-46f56af7-9f82-4d4f-9769-24023ce78e67.png)

**Comment by outdoorbits** [comment]
I just tried to change both parameters and always had the expected effect. Did you change these values and forced any display-output (by starting a backup)?

**Comment by estaban-zarzamora** [comment]
Several things tried to make the 128x32 more "readable":
1. changing font size, didn't improve. Size 12 is just best.
2. Changing the font in /var/www/oled.py from anything other then DejaVuSans.ttf doesn't work for me. Uploaded some other fonts to the pi in /usr/share/fonts/truetype/. Display would only work with: font = ImageFont.truetype('/usr/share/fonts/truetype/dejavu/DejaVuSans.ttf', 12)

Also tried to get a bigger SSH1106 OLED working instead of the SSD1306, but my knowledge falls short to properly edit /var/www/oled.py to make that work. Tried playing with disp = adafruit_displayio_sh1106.SH1106(128, 64, i2c,addr=I2C_ADDRESS)


Has anyone see other display integrations (hat, shield, bonnet) that allow for a bigger screen with better readability, easy and neat integration with a pi 3b+ and little code editing?  Maybe a LCD screen connected via SPI?

**Comment by outdoorbits** [comment]
Just to understand, why can't you use the standard display?

**Comment by estaban-zarzamora** [comment]
The 0.96" is to small for me to read without my reading glasses when on the road in Africa :-)

**Comment by outdoorbits** [comment]
Maybe there is a chance to enable ssd1309 devices also. But this might need some patience...

**Reply by estaban-zarzamora** [reply]
Thanks for thinking along. I am going for the 0.96 SSD1306 for now and just need to accept I need my glasses...

**Comment by outdoorbits** [comment]
I just uploaded a version supporting SPI-displays too. You will find everything in settings. There is no wiki yet but for connecting please have a look at https://luma-oled.readthedocs.io/en/latest/hardware.html
Does it work?
Please keep in mind, the output is prepared for displays 128x64, using 5 lines of text.

**Reply by estaban-zarzamora** [reply]
I will test it next week, thanks!

**Comment by rkosick** [comment]
I updated today and my display was not working anymore. Just keeping blank and dark. Using the 0.96 inch OLED SSD1306 Display I2C 128x64. Tried different settings but nothing works to get it running again.
I did restore my last backup and the display it works again.
Additional I now have  strange output even with my last backup. See attachment. 

[little-backup-box.txt](https://github.com/outdoorbits/little-backup-box/files/10874884/little-backup-box.txt)

But this is probably regarding connecting different lbb versions with the same browser.

**Reply by rkosick** [reply]
Did the update again still dark display. But after saving some settings in the browser the display shows up again and works fine.

**Reply by rkosick** [reply]
Somehow it is not active after booting anymore.

**Comment by outdoorbits** [comment]
Probably we are using the same display. Could you please compare your settings with mine:
![Screenshot_20230302_205606](https://user-images.githubusercontent.com/45213053/222538230-d56f6603-0b11-4ff0-8670-fed2db69f951.png)

**Reply by rkosick** [reply]
It's the same. Not working after booting. So I don't see the IP. When I connect and for example save a setting, it activates and works. Maybe I have to reinstall again? After Update I have the issue.

**Comment by rkosick** [comment]
I tried to reinstall all and its the same. Display is dark/not showing anything. Until I for example save settings.

**Comment by outdoorbits** [comment]
Just fixed a bug. Maybe after update you will to have save settings once again.

**Reply by rkosick** [reply]
Still the same. Display is blank after reboot.

**Comment by outdoorbits** [comment]
I found the issue and fixed it.

**Reply by rkosick** [reply]
I did an update right now and it is still not fixed for me. Display is not turning on after start. Only after I save the settings for example.

**Comment by outdoorbits** [comment]
I just uploaded some changes in update/installation process affecting the display-problem. I tested it in a new installation: The display worked without any configuration. This was reboot-proof. So please update again...

**Comment by rkosick** [comment]
![PXL_20230304_205639791~2](https://user-images.githubusercontent.com/122625940/222971703-c91ff158-d164-4cbd-8b1e-c9254274e73f.jpg)
![PXL_20230304_205637452~2](https://user-images.githubusercontent.com/122625940/222971707-0a5016c1-bcbd-47c8-92c1-56851d405723.jpg)
![PXL_20230304_205642837~2](https://user-images.githubusercontent.com/122625940/222971711-46bca93e-1958-414e-8f85-3d587b5afb54.jpg)

**Comment by outdoorbits** [comment]
This is your SH1106? Don't know where the leading dot comes from...

**Reply by rkosick** [reply]
I have the 1306. I will do a new installation and see what happens.

**Comment by outdoorbits** [comment]
I think I found where the dot came from. Just update...

**Reply by rkosick** [reply]
instead of the dot there is now a "1. " leading. (Without the quotation marks)

**Comment by outdoorbits** [comment]
Right, this means the first folder. In some cases cameras have more than one folder to backup, that's it.

**Reply by rkosick** [reply]
Ok sounds good. Works great now.

**Comment by estaban-zarzamora** [comment]
Yessss, I can read the 1.3"  sh1106 display without glasses!

**Comment by outdoorbits** [comment]
Great!!! Could you please provide a photo?

I just removed the "1." from screen in case of a single folder to backup.

**Reply by estaban-zarzamora** [reply]
![thumbnail_IMG_9845](https://user-images.githubusercontent.com/47415283/222982443-0668f550-b1d2-4d42-8b66-da6604f04699.jpg)
The 1.3" SH1106 OLED with the old 0.96" SSD1306 in the lid of the box in the background.

**Comment by rkosick** [comment]
Great job.

**Comment by outdoorbits** [comment]
Thank you.

Interesting case. Is it 3d printed?

**Reply by estaban-zarzamora** [reply]
It's a [Cyntech](https://shop.cyntech.co.uk/products/raspberry-pi-case-model-b-compatible) case, which can be used with 1 cm spacers to increase the height. This is perfect for my RPI 3b+ with [mSATA expansion board](https://geekworm.com/products/raspberry-pi-3-x850-v3-0-usb-3-0-msata-ssd-storage-expansion-board)
